vignettes/irecist.Rmd

```diff
@@ -0,0 +1,877 @@
+    ADT.join >= ADT + confirmation_period,
+  condition = AVALC == "iPR" &
+    all(AVALC.join %in% c("iCR", "iPR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1 &
+    (
+      min_cond(
+        var = ADT.join,
+        cond = AVALC.join == "iCR"
+      ) > max_cond(var = ADT.join, cond = AVALC.join == "iPR") |
+        count_vals(var = AVALC.join, val = "iCR") == 0 |
+        count_vals(var = AVALC.join, val = "iPR") == 0
+    ),
+  set_values_to = exprs(AVALC = "iPR")
+)
+```
+
+Please note that the result `AVALC = "Y"` for confirmed clinical benefit is
+defined by the first _two_ events specified for `events`. For subjects with
+observations fulfilling both events the one with the earlier date should be
+selected (and not the first one in the list).
+
+```{r}
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(icrsp_y_icr, icrsp_y_ipr, no_data_n),
+    set_values_to = exprs(
+      PARAMCD = "ICRSP",
+      PARAM = "Immune Confirmed Response by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(icrsp_y_icr, icrsp_y_ipr, icb_y, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "ICCB",
+      PARAM = "Immune Confirmed Clinical Benefit by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    ),
+    check_type = "none"
+  )
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr,ADT),
+    mode = "first",
+    events = list(icbor_icr, icbor_ipr, ibor_isd, ibor_non_icriupd, ibor_iupd, ibor_icpd, ibor_ne, no_data_missing),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "ICBOR",
+      PARAM = "Immune Best Confirmed Overall Response by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = aval_resp(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, RANDDT, ANL01FL),
+  filter = PARAMCD %in% c("ICRSP", "ICCB", "ICBOR")
+)
+```
+## Other Endpoints {#other}
+
+For additional endpoints that can be added please see [Creating ADRS (Including Non-standard Endpoints)](adrs.html).
```

Comment:
worth to mention which exactly parameters (not present here) are available there

Reply:
will update